packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -751,23 +789,37 @@ export class GarbageCollector implements IGarbageCollector {
             /** True to generate full GC data */
             fullGC?: boolean;
         },
-    ): Promise<IGCStats> {
-        const {
-            fullGC = this.gcOptions.runFullGC === true || this.summaryStateNeedsReset,
-        } = options;
-
+    ): Promise<IGCStats | undefined> {
+        const { fullGC = this.gcOptions.runFullGC === true || this.summaryStateNeedsReset } = options;
         const logger = options.logger
             ? ChildLogger.create(options.logger, undefined, { all: { completedGCRuns: () => this.completedRuns } })
             : this.mc.logger;
 
+        /**
+         * If there is no current reference timestamp, skip running GC. We need the current timestamp to track
+         * how long objects have been unreferenced and if they should be deleted.
+         *
+         * Note that the only scenario where GC is called and there is no reference timestamp is when no ops have ever
+         * been processed for this container and it is in read mode. In this scenario, there is no point in running GC
+         * anyway because references in the container do not change without any ops, i.e., there is nothing to collect.
+         */
+        const currentReferenceTimestampMs = this.runtime.getCurrentReferenceTimestampMs();
+        if (currentReferenceTimestampMs === undefined) {
```

Comment:
This check is in two different places. Have you considered whether or not currentReferenceTimestampMs should be passed into the collectGarbage function? There would be one less dependency on the runtime in a sense. Don't know if this is too expensive to do.

Reply:
It is used in 3 different places, and we would need to add this as a param to the constructor and 2 other APIs. The reason why this is an API on the runtime is because GC should have access to the current timestamp at all times.

Reply:
I thought the change would only be to the IGarbageCollector.collectGarbage method and the GarbageCollector.collectGarbage method and then the current reference timestamp would be set at the summarize layer. Granted this logic may be better to sit in GC's run GC method. This is a non-blocking comment.